api/query/atoms.go

```diff
@@ -257,7 +257,13 @@ func (t AbstractTriaged) BindToRuns(runs ...shared.TestRun) ConcreteQuery {
 	}
 	for _, run := range runs {
```

Comment:
Like you mentioned, this `triaged:test-issue` creates multiple `query.Triaged` objects, but in fact it is independent of `runs` information. In the [filter](https://github.com/web-platform-tests/wpt.fyi/blob/4aff68a58e7a7c90cef004eaf350a06a21f71993/api/query/cache/index/filter.go#L330), all it filters out is the testname. It means that if this `test-issue` exists in one run, it will exist in all runs. I would do something like this

```suggestion
        if t.Product == nil && len(runs) != 0 {
            metadata, _ := shared.GetMetadataResponse(make([]shared.TestRun, 0), logrus.StandardLogger(), t.metadataFetcher)
	    metadataMap := shared.PrepareLinkFilter(metadata)
             return Or{Triaged{runs[0].ID, metadataMap}}
       }
	for _, run := range runs {
```

let me know if it makes sense